torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -161,6 +162,7 @@ def _pre_forward(
             args, kwargs = self._fsdp_param_group.pre_forward(module, args, kwargs)
         return args, kwargs
 
+    @torch.compiler.disable_if_config_true("skip_fsdp_hooks")
```

Comment:
Do you plan to use this API for anything apart from `skip_fsdp_hooks`? If not, I would suggest to just define the decorator at the top of this file. And just keep it really simple

```
def disable_if_config_true(func):
        @functools.wraps(func)
        def inner(*args, **kwargs):
            if config.skip_fsdp_hooks:
                return disable(func)(*args, **kwargs)
            else:
                return func(*args, **kwargs)

        return inner
```